### PR TITLE
[Cppcheck] Support the option `-I`

### PR DESCRIPTION
**Pull request by suetsuguk**
Related to:
- https://github.com/sider/runners/issues/1280

### CODE REVIEW

File: docs/getting-started/custom-configuration.md [hunk 1]
```diff
@@ -228,6 +228,54 @@ When the `npm_install` option is not `false`, Sider will try as follows:
 When the option is `false`, Sider will skip these installation steps and analyze with the pre-installed default version.
 You might want to set the option to `false` if you don't configure analyzer and don't want to see warnings.
 
+## `linter.<analyzer_id>.apt`
+
+_Type:_ `string`, `string[]`
+
+Some projects particularly written in C/C++ depend on development packages. Sider can install packages with the `apt` package manager, which is on `Debian` based Linux distributions.
```
Comment (yykamei) [inline docs/getting-started/custom-configuration.md:235]:
```suggestion
Development packages provided by the OS environment may be necessary, particularly for projects written in C/C++. Sider lets you install packages with APT, which is a package manager for `Debian` based Linux distributions.
```

File: docs/getting-started/custom-configuration.md [hunk 2]
```diff
@@ -228,6 +228,54 @@ When the `npm_install` option is not `false`, Sider will try as follows:
...
+```yaml
+linter:
+  clang_tidy:
+    include-path:
+      - myinclude
+      - foo/include
+      - /usr/include/libfastjson
+```
+
+Sider treats this option as a compilation option `-I` and passes it to the `clang-tidy` command internally as:
```
Comment (ybiquitous) [inline docs/getting-started/custom-configuration.md:271]: [suggestion] Users may be confused for `clang-tidy` if they want to use `cppcheck`. How about rewriting it as follows? 👇 

```suggestion
Sider treats this option as a compilation option `-I` and passes it to the analysis tool command internally as:
```

File: docs/getting-started/custom-configuration.md [hunk 3]
```diff
@@ -228,6 +228,54 @@ When the `npm_install` option is not `false`, Sider will try as follows:
...
+_Type:_ `string`, `string[]`
+
+Some projects particularly written in C/C++ depend on development packages. Sider can install packages with the `apt` package manager, which is on `Debian` based Linux distributions.
+
+The `apt` option allows you to specify a list of development packages your project depends on.
+The packages must satisfy the conditions below:
+
+- Packages compatible with [our Docker image](https://github.com/sider/devon_rex/blob/master/base/Dockerfile).
+- Packages with "-dev" suffix in names are available.
+- A specific version number can be requested with the `<name>=<version>` format.
```
Comment (yykamei) [inline docs/getting-started/custom-configuration.md:242]: How about this? I think these conditions *must* be applied for the `apt` options.

```suggestion
- Packages must be compatible with [our Docker image](https://github.com/sider/devon_rex/blob/master/base/Dockerfile).
- Package names must be suffixed with "-dev".
- Each package name must be formatted as `<name>` or `<name>=<version>`.
```

File: docs/getting-started/custom-configuration.md [hunk 4]
```diff
@@ -228,6 +228,54 @@ When the `npm_install` option is not `false`, Sider will try as follows:
...
+    apt:
+      - libgdbm-dev
+      - libfastjson-dev=0.99.8-2
+```
+
+## `linter.<analyzer_id>.include-path`
+
+_Type:_ `string`, `string[]`
+
+Some C/C++ analyzers can handle include paths like any other C/C++ preprocessor. With Sider, the `include-path` option allows you to add directories to include search path.
```
Comment (yykamei) [inline docs/getting-started/custom-configuration.md:258]: I thought *analyzers* and *preprocessors* are not in the same category, so I removed "any other."

```suggestion
Some C/C++ analyzers can handle include paths like C/C++ preprocessors can. With Sider, the `include-path` option allows you to add directories to include search paths.
```

File: docs/getting-started/custom-configuration.md (hunk 2 above)
Comment (yykamei) [inline docs/getting-started/custom-configuration.md:271]: This section is not specific for Clang-Tidy, so I edited here like this:

```suggestion
Sider treats this option as a compilation option `-I` and passes it as command-line arguments internally. For example, [Clang-Tidy](../tools/cplusplus/clang-tidy.md) treats this option like this:
```

Comment (suetsuguk) [reply docs/getting-started/custom-configuration.md:271]: Agreed. But I will adopt another suggestion at https://github.com/sider/sider-docs/pull/476#discussion_r457905229

File: docs/getting-started/custom-configuration.md (hunk 1 above)
Comment (suetsuguk) [reply docs/getting-started/custom-configuration.md]: Agreed. commit 68b002999a1ee1fb86b7471e74738d0b9c6d54d8

File: docs/getting-started/custom-configuration.md (hunk 3 above)
Comment (suetsuguk) [reply docs/getting-started/custom-configuration.md]: Agreed. commit 68b002999a1ee1fb86b7471e74738d0b9c6d54d8

File: docs/getting-started/custom-configuration.md (hunk 4 above)
Comment (suetsuguk) [reply docs/getting-started/custom-configuration.md]: Agreed. commit 68b002999a1ee1fb86b7471e74738d0b9c6d54d8

File: docs/getting-started/custom-configuration.md (hunk 2 above)
Comment (suetsuguk) [reply docs/getting-started/custom-configuration.md]: Agreed. commit 68b002999a1ee1fb86b7471e74738d0b9c6d54d8

File: docs/tools/cplusplus/cppcheck.md [hunk 5]
```diff
@@ -38,23 +38,25 @@ linter:
...
-| [`std`](#std)                                                                         | `string`             | -       |
-| [`project`](#project)                                                                 | `string`             | -       |
-| [`language`](#language)                                                               | `string`             | -       |
-| [`addon`](#addon)                                                                     | `string`, `string[]` | -       |
-| [`bug-hunting`](#bug-hunting)                                                         | `boolean`            | `false` |
+| Name                                                                                          | Type                 | Default |
+| --------------------------------------------------------------------------------------------- | -------------------- | ------- |
+| [`root_dir`](../../getting-started/custom-configuration.md#linteranalyzer_idroot_dir)         | `string`             | -       |
+| [`include-path`](../../getting-started/custom-configuration.md#linteranalyzer_idinclude-path) | `string`, `string[]` | _       |
```
Comment (yykamei) [inline docs/tools/cplusplus/cppcheck.md:51]: Is this a typo?

```suggestion
| [`include-path`](../../getting-started/custom-configuration.md#linteranalyzer_idinclude-path) | `string`, `string[]` | -       |
```

File: docs/tools/cplusplus/clang-tidy.md [hunk 6]
```diff
@@ -37,28 +37,10 @@ linter:
...
-| ------------------------------------------------------------------------------------- | -------------------- | ------------- |
-| [`root_dir`](../../getting-started/custom-configuration.md#linteranalyzer_idroot_dir) | `string`             | -             |
-| [`apt`](#apt)                                                                         | `string`, `string[]` | `[]`          |
-| [`include-path`](#include-path)                                                       | `string`, `string[]` | _(See below)_ |
+| Name                                                                                          | Type                 | Default |
+| --------------------------------------------------------------------------------------------- | -------------------- | ------- |
+| [`root_dir`](../../getting-started/custom-configuration.md#linteranalyzer_idroot_dir)         | `string`             | -       |
+| [`apt`](../../getting-started/custom-configuration.md#linteranalyzer_idapt)                   | `string`, `string[]` | -       |
+| [`include-path`](../../getting-started/custom-configuration.md#linteranalyzer_idinclude-path) | `string`, `string[]` | _       |
```
Comment (yykamei) [inline docs/tools/cplusplus/clang-tidy.md:44]:
```suggestion
| [`include-path`](../../getting-started/custom-configuration.md#linteranalyzer_idinclude-path) | `string`, `string[]` | -       |
```

File: docs/tools/cplusplus/cppcheck.md (hunk 5 above)
Comment (suetsuguk) [reply docs/tools/cplusplus/cppcheck.md]: Thank you for pointing out. commit 578c8cb385f5f3fb246090ef002097595ccc7ac8

File: docs/tools/cplusplus/clang-tidy.md (hunk 6 above)
Comment (suetsuguk) [reply docs/tools/cplusplus/clang-tidy.md]: Thank you for pointing out. commit 578c8cb385f5f3fb246090ef002097595ccc7ac8

**Comment by ybiquitous** [comment]
When the new Runners 0.31.0 is released, I will merge this PR.

https://github.com/sider/runners/releases/tag/0.31.0

**Pull request merged**